pkgs/by-name/bt/btrfs-list/package.nix

```diff
@@ -0,0 +1,38 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  btrfs-progs,
+  perl,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "btrfs-list";
+  version = "2.3";
+
+  src = fetchFromGitHub {
+    owner = "speed47";
+    repo = "btrfs-list";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-cWzDRop0cyrjVIJzuZxTqELgec66GiPAUJY1xIBr3OY=";
+  };
+
+  buildInputs = [ perl ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    install -D -t $out/bin btrfs-list
+
+    wrapProgram $out/bin/btrfs-list --prefix PATH : ${lib.makeBinPath [ btrfs-progs ]}
+  '';
+
+  meta = with lib; {
```

Comment:
Also remove `with lib`, thanks

Reply:
😭 ok but why? Is there some consensus on when it’s a bad/good idea? Seems pretty fine to me in this case.

Reply:
The `with` keyword expands its scope with a low priority. You need to use evaluation to know which identifiers there are. This is a general problem, so we try to avoid using it on a large scale as much as possible. Small-scale use, neither top-level, nested, or escaping with, is still acceptable.

The most special problem here is when many people tried to migrate the package from using `rec` to the `finalAttrs` pattern, they ignored `with lib; version` in `changelog`, causing it to become lib.version, which already shows this pattern is misleading.

Reply:
> The most special problem here is when many people tried to migrate the package from using rec to the finalAttrs pattern, they ignored with lib; version in changelog, causing it to become lib.version, which already shows this pattern is misleading.

Do you have an example of this? As I don't fully understand what you mean.

Anyway, I'll make the change, as I trust you that it's been discussed and agreed upon not to use it in this case (although it's still used for `meta` extensively in nixpkgs, and I don't know if anyone is planning on removing it en masse?)